rest_framework/permissions.py

```diff
@@ -112,7 +112,8 @@ def has_permission(self, request, view):
         except AttributeError:
             queryset = getattr(view, 'queryset', None)
         except AssertionError:
```

Comment:
I'm not sure why we have an `AssertionError` block here, but it looks wrong. Just the first four lines are sufficient.

    try:
        queryset = view.get_queryset()
    except AttributeError:
        queryset = getattr(view, 'queryset', None)